src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -138,6 +208,19 @@ public static String encodeSha256Hash(byte[] data) {
             bytes[i] = (byte) encoded[i];
         }
 
+        int leftoverBits = (bytes.length * 5) % 8;
+        if (leftoverBits > 0) {
```

Comment:
If `leftoverBits` >= 5 this check would be wrong, because you would be checking a full leftover char, which shouldn't had been there at all.

Reply:
e.g. `MHA` would pass the check, but it should fail because, there are (3*5)%8  = 7 leftover bits, so:

1. `A` shouldn't be there (because it provides 5 unused bits)
2. That leaves 2 extra unused bits in `H`. `H` in base 32 is `00111` , which isn't a proper canonical representation, it should be `00100` (`E`), because the last two bits should be 0.

Reply:
Note also, that, while all these checks are needed in Golang (because the decoder in `"encoding/base32"` swallows non-canonical base32 strings) the Java library maybe be able to do detect this without extra code.

Reply:
`guava` doesn't seem to provide that functionality. There may be other libraries that do, though